packages/bson-transpilers/symbols/php/templates.yaml

```
@@ -0,0 +1,1231 @@
+        (pattern, flags) => {
+            // This is some standalone object context, which is generated to parse 
+            // node and doesn't have access to main Generator object. Thus we can't
+            // use utility function call. All ~Template calls use this type of 
+            // context. All ~ArgsTemplate have access to utility functions.
+
+            stringifyWithSingleQuotes = (str) => {
+                if (
+                    (str.charAt(0) === '\'' && str.charAt(str.length - 1) === '\'') ||
+                    (str.charAt(0) === '"' && str.charAt(str.length - 1) === '"')
+                ) {
+                    str = str.substr(1, str.length - 2);
+                }
+                return `${str.replace(/\\([\s\S])|(')/g, '\\$1$2')}`;
+            };
+        
+            return `'/${stringifyWithSingleQuotes(pattern)}/${flags ? flags : ''}'`;
+        }
+    BoolTypeTemplate: &BoolTypeTemplate !!js/function >
+        (literal) => {
+            return literal.toLowerCase();
+        }
+    IntegerTypeTemplate: &IntegerTypeTemplate null # args: literal, argType (i.e. '_string', '_decimal' etc)
+    DecimalTypeTemplate: &DecimalTypeTemplate null # args: literal, argType
+    LongBasicTypeTemplate: &LongBasicTypeTemplate null # args: literal, argType
+    HexTypeTemplate: &HexTypeTemplate null # args: literal, argType
+    OctalTypeTemplate: &OctalTypeTemplate !!js/function >
+        (literal) => {
+            let offset = 0;
+
+            if (
+                literal.charAt(0) === '0' &&
+                (literal.charAt(1) === '0' || literal.charAt(1) === 'o' || literal.charAt(1) === 'O')
+            ) {
+                offset = 2;
+            } else if (literal.charAt(0) === '0') {
+                offset = 1;
+            }
+
+            literal = `0${literal.substr(offset, literal.length - 1)}`;
+
+            return literal;
+        }
+    NumericTypeTemplate: &NumericTypeTemplate null # args: literal, argType
+    ArrayTypeTemplate: &ArrayTypeTemplate !!js/function >
+        (literal, depth) => {
+            if (literal === '') {
+                return '[]'
+            }
+            return `[${literal}]`;
+        }
+    ArrayTypeArgsTemplate: &ArrayTypeArgsTemplate null # Args: single array element, nestedness, lastElement? (note: not being used atm)
+    NullTypeTemplate: &NullTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    UndefinedTypeTemplate: &UndefinedTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    ObjectTypeTemplate: &ObjectTypeTemplate !!js/function >
+        (literal) => {
+            return `${literal}`;
+        }
+    ObjectTypeArgsTemplate: &ObjectTypeArgsTemplate !!js/function >
+        (args, depth) => {
+            let isObj = true;
+        
+            if (args.length === 0) {
+                return `[]`;
+            }
+
+            const checkIndex = (strIndex, numericIndex) => {
```

Suggesting the following rename to make the code more self-documenting. If I understand correctly, this is just checking if the actual `strIndex` is equivalent to an expected `numericIndex`.

```suggestion
            const isExpectedIndex = (actualIndex, expectedIndex) => {
```

Feel free to correct this if I'm misunderstanding what this function actual does, but either way I think the naming can be improved here. I'd definitely like to see "actual" and "expected" in the parameter names.